lib/child_process.js

```diff
@@ -818,6 +820,10 @@ function spawnSync(file, args, options) {
 
   debug('spawnSync', options);
 
+  // Validate the arguments for null bytes. The normalizeSpawnArguments()
+  // function already makes sure that the array is present.
+  validateArgumentsNullCheck(options.args);
```

Comment:
It'd be good to also apply this to file, shell, cwd, env, etc. The exec functions should also check for nul bytes.